testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -3297,6 +3297,28 @@ static void checkOverlaps(OverlapChecker c) {
         DECIMAL_OVERFLOW, true);
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6706">[CALCITE-6706]
+   * Checked arithmetic does not take effect in subqueries</a>. */
+  @Test void testCastOverflow() {
+    final SqlOperatorFixture f = fixture().withConformance(SqlConformanceEnum.BIG_QUERY);
+    f.checkFails("SELECT -CAST(-32768 AS SMALLINT)",
+        ".*Value 32768 does not fit in a SMALLINT", true);
```

Comment:
`.*` is necessary for this test?

Reply:
That takes the place of the substring "integer overflow:" I have replaced the * with this string in the test.